OpenRA.Mods.Common/Traits/World/Pathfinder/Graph.cs

```diff
@@ -0,0 +1,187 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public interface IGraph<T> : IDisposable
```

Comment:
Yes but this is specific to pathing (or so it seems), while the name does not suggest that.